app/web/src/features/messages/utils.ts

```diff
@@ -58,3 +58,19 @@ export function groupChatTitleText(
         .map((user) => firstName(user?.name))
         .join(", ");
 }
+
+/** Returns the other user's username, or null if there are more than 2 users. */
+export function formatGroupChatUsernames(
```

Comment:
> groupChatTitleText format the name Good but it does not takes the username used for the url /user/{username} ? Or maybe i can use the same function and add a parameters to the function like returnUsername as boolean ?

Oops first of all, I didn't read the comment above the function/code properly until now, sorry! 🙈  But it looks like this is only intended for "group chats" with exactly one other user (i.e. it's a DM), so maybe it's better to name the function `formatUserInDm` or something similar to those words.

And I see what you mean about the problem of not being able to get the username back out. Instead of separating the formatting of the text and the JSX separately, I think maybe it's worth combining them into the `groupChatTitleText` helper function above.

So I mean something like the following (rough code to give you an idea, so might not work exactly!):

```tsx
export function groupChatTitleText(
  groupChat: GroupChat.AsObject,
  groupChatMembersQuery: {
    data?: Map<number | undefined, User.AsObject | undefined>;
  },
  currentUserId: number
) {
  if (groupChat.title) {
    return <PageTitle>{groupChat.title}</PageTitle>;
  }

  if (groupChatMembersQuery.isLoading) {
    return <PageTitle>Chat</PageTitle>;
  }

  const otherGroupChatMembers = Array.from(
    groupChatMembersQuery.data?.values() ?? []
  ).filter((user) => user?.userId !== currentUserId);

  // DM
  if (otherGroupChatMembers.length === 1) {
    return (
      <Link>
        <PageTitle>{firstName(user?.name)}</PageTitle>
      </Link>
    );
  }

  // group chat
  return (
    <PageTitle>
      {otherGroupChatMembers.map((user) => firstName(user?.name)).join(", ")}
    </PageTitle>
  );
}
```

Reply:
I understand and made those changes, in order to achieve this i had to replace the file `utils.ts` to `utils.tsx`, and i changes parts of the code that uses that function.

There is one remaining problem tho, line 190 on file `GroupChatView.tsx` we make use of title ( which is the result of the above function ) as title for HtmlMeta, function which now returns JSX, do you have any idea how i could solve this in a clean way ? 

Also this causes the GroupChatListItem a red warning as it makes nesting of two `a` element as follow 
`<a><a> </a></a>`. ^^'

One idea i have is keeping the code as you suggested above, and adding a fourth parameters `withoutJSX` as a boolean to either get back the JSX or the string, default to false ?

Reply:
PS: Sorry for delay the end of the year was busy ^^'

Reply:
Ahh I didn't see the `title` being used for the HTML metadata as well, what a pain in the arse!

In that case, I think what you got there already is probably "cleaner" than what I suggested, as I think mixing the helper to sometimes return JSX, and at other times return text only is more confusing, since the function is doing too much imo at that point.